### PR TITLE
Adding search history

### PR DESCRIPTION
**Pull request by xakiru**
Hey guys,
Adding the search history, with the possibility to enable/disable it from settings.
ref #185
Thanks,

### CODE REVIEW

File: default.py [hunk 1]
```diff
@@ -220,13 +220,37 @@ def PLANNED(payload, params):
...
+    history = control.getSetting("9anime.history")
+    history_array = history.split(history_delimiter)
+    if history != "" and "Yes" in control.getSetting('searchhistory') :
+        return control.draw_items(_BROWSER.search_history(history_array))
+    else :
+        return SEARCH(payload,params)
+
 @route('search')
 def SEARCH(payload, params):
+    history_delimiter=":_:"
```
Comment (DxCx) [inline default.py]: can we define this globally?
HISTORY_DELIM

File: default.py [hunk 2]
```diff
@@ -220,13 +220,37 @@ def PLANNED(payload, params):
...
+    history = control.getSetting("9anime.history")
+    history_array = history.split(history_delimiter)
     query = control.keyboard(control.lang(30007))
     if query:
+        if history != "" :
+            query = query+history_delimiter
+        history=query+history
+        while history.count(history_delimiter) > 6 :
+            history=history.rsplit(history_delimiter, 1)[0]
+        control.setSetting("9anime.history",history)
```
Comment (DxCx) [inline default.py]: why would you handle history if `not control.getSetting('searchhistory')` ?

File: resources/lib/NineAnimeBrowser.py [hunk 3]
```diff
@@ -169,6 +173,12 @@ def search_site(self, search_string, page=1):
         url = self._to_url("search")
         return self._process_anime_view(url, data, "search/%s/%%d" % search_string, page)
 
+    def search_history(self,search_array):
+    	result = map(self._parse_history_view,search_array)
+    	result.insert(0,utils.allocate_item("New Search", "search", True))
+    	return result
+        #all_results += self._handle_paging(results, base_plugin_url, page)
```
Comment (DxCx) [inline resources/lib/NineAnimeBrowser.py]: cleanup :D

File: default.py [hunk 4]
```diff
@@ -220,13 +220,37 @@ def PLANNED(payload, params):
...
         return control.draw_items(_BROWSER.search_site(query))
-    return False
+    return False   
 
+@route('clear')
```
Comment (DxCx) [inline default.py]: where is the link for that?
havn't seen one..

File: default.py (hunk 1 above)
Comment (xakiru) [reply default.py]: Sure thing

File: default.py (hunk 2 above)
Comment (xakiru) [reply default.py]: you're right, let me fix this one.

File: resources/lib/NineAnimeBrowser.py (hunk 3 above)
Comment (xakiru) [reply resources/lib/NineAnimeBrowser.py]: lol yea

File: default.py (hunk 4 above)
Comment (xakiru) [reply default.py]: I was using a button to clean the history.
Hmm is there a function to go back to the main plugin window ?

Comment (xakiru) [reply default.py]: I'm putting back the clear button, it's in the bottom of the history list.

**Comment by xakiru** [comment]
I made the requested changes.
Thanks

**Comment by DxCx** [comment]
hey @xakiru havn't forgotten this, im waiting for playback to be stable again before merging another feature ;)

**Comment by xakiru** [comment]
Yea sure, take your time.

**Comment by DxCx** [comment]
Thank you very much @xakiru 
and thanks for the patience :+1:

**Pull request merged**